tests/acceptance/bootstrap/SpacesContext.php

```diff
@@ -2062,13 +2062,12 @@ public function userHasCopiedOrMovedFileInsideSpaceUsingFileId(
 	): void {
 		// split the source when there are sub-folders
 		$sourceFile = \trim($sourceFile, "/");
-		$sourceFile = \explode("/", $sourceFile);
-		$sourceFile = \end($sourceFile);
 		$destinationFile = \trim($destinationFile, "/");
-		if ($actionType === 'copied' || $actionType === 'moved') {
-			$fileDestination = $this->escapePath($destinationFile) . '/' . $this->escapePath($sourceFile);
-		} else {
-			$fileDestination = $destinationFile;
+
+		$fileDestination = $this->escapePath($destinationFile);
+
+		if ($actionType !== 'renamed' && !strpos($destinationFile, "/")) {
```

Comment:
this generally looks good but IMO we better have separate step for rename action

Reply:
Yeah. Sounds better.. Did it just now